src/Shared/ChangeWaves.cs

```diff
@@ -88,121 +105,81 @@ internal static ChangeWaveConversionState ConversionState
-            if (!AllWavesAsVersion.Contains(changeWave))
-            {
-                foreach (Version wave in AllWavesAsVersion)
-                {
-                    if (wave > changeWave)
-                    {
-                        ConversionState = ChangeWaveConversionState.Valid;
-                        DisabledWave = wave.ToString();
-                        return;
-                    }
-                }
-            }
-
+            // _cachedWave is somewhere between valid waves, find the next valid version.
+            _cachedWave = AllWaves.Where((x) => x > _cachedWave).First();
             ConversionState = ChangeWaveConversionState.Valid;
-            DisabledWave = changeWave.ToString();
         }
 
         /// <summary>
-        /// Compares the passed wave to the MSBuildDisableFeaturesFromVersion environment variable.
-        /// Version MUST be of the format: "xx.yy".
+        /// Determines whether features behind the given wave are enabled.
         /// </summary>
         /// <param name="wave">The version to compare.</param>
-        /// <returns>A bool indicating whether the feature behind a version is enabled.</returns>
-        public static bool AreFeaturesEnabled(string wave)
-        {
-            Version waveToCheck;
-
-            // When a caller passes an invalid wave, fail the build.
-            ErrorUtilities.VerifyThrow(Version.TryParse(wave.ToString(), out waveToCheck),
-                                       $"Argument 'wave' passed with invalid format." +
-                                       $"Please use pre-existing const strings or define one with format 'xx.yy");
-
-            return AreFeaturesEnabled(waveToCheck);
-        }
-
-        /// <summary>
-        /// Compares the passed wave to the MSBuildDisableFeaturesFromVersion environment variable.
-        /// </summary>
-        /// <param name="wave">The version to compare.</param>
-        /// <returns>A bool indicating whether the version is enabled.</returns>
+        /// <returns>A bool indicating whether the change wave is enabled.</returns>
         public static bool AreFeaturesEnabled(Version wave)
         {
-            if (_state == ChangeWaveConversionState.NotConvertedYet)
+            if (ShouldApplyChangeWave)
             {
                 ApplyChangeWave();
             }
 
-            // This is opt out behavior, all waves are enabled by default.
-            if (DisabledWave.Length == 0 || DisabledWave.Equals(EnableAllFeatures, StringComparison.OrdinalIgnoreCase))
-            {
-                return true;
-            }
-
-            Version currentSetWave;
-
-            // If we can't parse the environment variable, default to enabling features.
-            if (!Version.TryParse(DisabledWave, out currentSetWave))
+            // Check if we cached the special value to enable all features behind change waves.
+            if (_cachedWave == EnableAllFeatures)
```

Comment:
This is captured by the next statement, right?

Reply:
You're absolutely correct 🤦‍♂️